tests/IPv6NetworkSpecTest.php

```diff
@@ -63,6 +63,18 @@ public function testContainsWorksWithIPv4(): void
         self::assertFalse($network->contains(IPv4Address::parse('127.0.0.1')));
     }
 
+    public function testContainsWorksWithNetworkInNetwork(): void
+    {
+        $network = IPv6Network::parse('2001:db8::/64');
+
+        self::assertTrue($network->contains(IPv6Network::parse('2001:db8::/64')));
+        self::assertTrue($network->contains(IPv6Network::parse('2001:db8::42/122')));
+        self::assertTrue($network->contains(IPv6Network::parse('2001:db8::43/128')));
+//
+        self::assertFalse($network->contains(IPv6Network::parse('2001:db9::/64')));
+        self::assertFalse($network->contains(IPv6Network::parse('2001:db8::/48')));
+    }
```

Comment:
Could you add a test case for checking on IPv4 networks. An IPv6 network should never contain an IPv4 network

We can also track this in a separate issue if you want.